onnxscript/optimizer/_constant_folding.py

```diff
@@ -1087,6 +1087,23 @@ def visit_graph(self, graph: ir.Graph) -> None:
         for node in graph:
             self.visit_node(node, graph)
 
+        # Replace outputs if output nodes can be folded. This are typically outputs from
+        # Identity nodes
+        for i, output in enumerate(graph.outputs):
+            if output is None:
+                continue
+            sym_value = self._state.get_sym_value(output)
+            if not isinstance(sym_value, ir.Value):
+                # An output must be a Value
+                continue
+            if sym_value in graph.inputs:
+                # ONNX does not allow a graph output to be a graph input
+                continue
+            # Rename sym_value to match the output name
```

Comment:
I just realized that there is another edge case: we can't do this for two different outputs: eg.
```
   x = Add(...)
   graph_output1 = Identity(x)
   graph_output2 = Identity(x)
```
We can rename x to either graph_output1 or graph_output2, not both.

Reply:
Thanks - updated

Reply:
Added a test

Reply:
Looks like the subgraphs are still not handled correctly: https://github.com/microsoft/onnxscript/actions/runs/14414536354/job/40428982831?pr=2180#step:6:809

Reply:
Fixed